cmd/broker/filter/main.go

```diff
@@ -147,11 +148,12 @@ func main() {
 	}
 
 	reporter := filter.NewStatsReporter(env.ContainerName, kmeta.ChildName(env.PodName, uuid.New().String()))
+	authReporter := filter.NewAuthStatsReporterAdapter(reporter)
```

Comment:
Creating an interface to avoid import cycle.

Before this, it is stuck with a import cycle between `auth` and `filter` packages. The `auth` code needs `filter`'s stats reporting, but `filter` is already using `auth` stuff. To fix this I created a simple interface in `auth` that just covers the stats we need. Then making an adapter in `filter` to bridge the gap. Is this a good approach to solve this problem?